App/Redux/PreferencesRedux.ts

```diff
@@ -46,8 +46,8 @@ export type PreferencesState = {
   publicKey?: string
   profile?: Profile
   pending?: string,
-  readonly services: {[index: string]: Service}
-  readonly tourScreens: {[index: string]: boolean} // true = still need to show, false = no need
+  readonly services: {[k in ServiceType]: Service}
+  readonly tourScreens: {[k in TourScreens]: boolean} // true = still need to show, false = no need
```

Comment:
This enforces that only certain keys are used. Actually in this case it also ensures that **all** keys are used. If you wanted to ensure that the keys that are used are part of a known set, but you didn't want to require that all the keys are used, you could do:
```
readonly services: {[k in ServiceType]?: Service}
```